### PR TITLE
Eth updates

### PR DESCRIPTION
**Pull request by mossberg**
- add human_transactions top level function to ManticoreEVM
- allow str types to be used in `solidity_create_contract(..., args=('str goes here'))` (not only bytes)
- automatically truncate symbols that are too big for the solidity function argument they correspond to. e.g. passing a `BitVec(256)` into a function argument that takes an `address` (160 bits). previously we would assertion fail in ZEXTEND
- refactor ABI._readBE. add an offset parameter.  fix a logic bug.
- rename EVMAccount.name to EVMAccount.name_ (underscore at the end). no member variables of EVMAccount/EVMContract should collide with variable/function names commonly expected to be seen in Solidity contracts, because we bail out. `name` can reasonably be expected to be the name of a Solidity state variable, so we don't want to collide with this.

---
This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/1195)

### CODE REVIEW

File: manticore/ethereum/__init__.py [hunk 1]
```diff
@@ -626,6 +626,11 @@ def transactions(self, state_id=None):
         state = self.load(state_id)
         return state.platform.transactions
 
+    def human_transactions(self, state_id=None):
+        """ Transactions list for state `state_id` """
```
Comment (feliam) [inline manticore/ethereum/__init__.py]: If id is None and there is more than a single running state it raises an exception. (or it should)

File: manticore/ethereum/abi.py [hunk 2]
```diff
@@ -129,8 +129,10 @@ def _serialize_bytes(value):
         Serializes the value and pads to multiple of 32 bytes
 
         :param value:
-        :type value: bytearray or Array
+        :type value: str or bytearray or Array
         """
+        if isinstance(value, str):
```
[the remark this answers is not in the document]
Comment (mossberg) [reply manticore/ethereum/abi.py]: ok! moved it up.

[the remark this answers is not in the document]
Comment (mossberg) [reply manticore/ethereum/abi.py]: ok! moved it up.

[the remark this answers is not in the document]
Comment (mossberg) [reply manticore/ethereum/abi.py]: ok! moved it up.

**Comment by mossberg** [comment]
this should be good to go! just waiting on another look from @feliam but i'd like to merge this soon.

**Pull request merged**